nntrainer/tensor/tensor.cpp

```diff
@@ -1088,6 +1105,46 @@ void Tensor::setTensorVar(TensorDim d, void *buf, size_t offset) {
   itensor->setTensorVar(d, buf, offset);
 }
 
+TensorDim Tensor::calculateConcatOutputDim(const std::vector<Tensor> &tensors,
+                                           int axis) {
```

Comment:
This newly added private static function is to calculate the output tensor dimension with tensor vectors and axis.
It is used for checking the given output tensor dimension and creating a new output tensor if not provided.